lib/internal/child_process.js

```diff
@@ -478,13 +482,16 @@ function setupChannel(target, channel) {
         // read because SCM_RIGHTS messages don't get coalesced. Make sure
         // that we deliver the handle with the right message however.
         if (isInternal(message)) {
-          if (message.cmd === 'NODE_HANDLE')
-            handleMessage(message, recvHandle, true);
-          else
+          if (message.cmd === 'NODE_HANDLE') {
+            handleMessage(message, pendingHandle, true);
+          } else {
             handleMessage(message, undefined, true);
+          }
         } else {
           handleMessage(message, undefined, false);
         }
+
+        pendingHandle = null;
```

Comment:
This looks incorrect: it clears the variable on the first iteration but what if the NODE_HANDLE message is not in the first chunk?  I would have expected to see this assignment in the `if (message.cmd === 'NODE_HANDLE') {` block.